app/database_etl/location_utils/location_util_functions.py

```diff
@@ -94,7 +96,28 @@ def row_in_feature_layer(row: pd.Series, feature_layer: FeatureLayer) -> bool:
     pin = Point({"x": row['pin_longitude'], "y": row['pin_latitude']})
     # construct a geometry filter to check if each point is in a disputed area
     pin_filter = intersects(pin)
-    in_disputed_area = len(feature_layer.query(geometry_filter=pin_filter).features) > 0
+
+    continue_query = True
+    retries = 0
+    MAX_RETRIES = 9
+    # Default to setting in_disputed_area = True to ensure we never show pins in disputed area
+    in_disputed_area = True
+    # Make query to determine whether or not the pin is in the disputed area
+    # If the query times out, retry with exponential backoff
+    while continue_query:
+        try:
+            in_disputed_area = len(feature_layer.query(geometry_filter=pin_filter).features) > 0
+            continue_query = False
+        except Exception as e:
+            # send slack message if we exceed retry count
+            if retries > MAX_RETRIES:
+                body = f'Unable to check if the record with ID {row["source_id"]} is in a disputed region.'
+                send_slack_message(body, channel='#dev-logging-etl')
+                continue_query = False
+            else:
+                sleep(1.5**(retries))
```

Comment:
Cognizant that this may slow down ETL considerably if we get a series of these errors. This permits 1 minute per record -- which would mean the whole ETL would take over 100+ hours to run if all records error. Thoughts on the rate of backoff / bounds of backoff here?

Reply:
Yeah I'm worried about speed as well - I put relatively high backoff to be a bit more conservative because I wasn't sure how long the arcgis server would need to recover in the event of a failure. I can try using smaller backoff (maybe like 1.2**retries) instead and then adjust if we continue to get less reliable etl runs.